### ISSUE TITLE
"WebSend":"Connect failed" for status update ALMOST always >> EDIT: Checked Tasmota firmware... problem begins when pairing with Smartthings!

### ISSUE THREAD

**Issue by dcuche** [hongtat/tasmota-connect#18]
Hello,
Just flashed tasmota 9.1 on a Sonoff TX 1 CH wall switch, and added to Smartthings HUB 2 with Tasmota Connect... but..
ALMOST every time (like 75% of the times) that the swtich is pressed, either physically, through the app or the tasmota web UI, the switch hangs for like 5 seconds until getting a timeout with the error 
17:28:00 RSL: stat/tasmota_D572A8/RESULT = {"WebSend":"Connect failed"}
(for example).
Like, it cannot connect to the hub for status update...
But, it does not happens always... sometimes it works... so, strange.

Anyone with any similar issue?

Thank you!

EDIT: I tried flashing again Tasmota in the sonoff switch... the stitch works perfectly with tasmota via web UI and physically... BUT... only after pairing it with Tasmota-connect with smartthings... the problems begin (WebSend: Connection Failed).

Bottom line: Now even when physically pressing the button, the switch can only take inputs every 5 seconds (you cannot turn the light on and off again within 5 seconds.. the device does not respond)

**Comment by dcuche** [comment]
Here are some more console details:

First I turned it ON (from the smartthings app) and it failed to update back (the switch then freezes for 5 seconds... ) and then when I turned it OFF, it worked right away.
but.... like 75% of the times if does not work... and when it works... it works with on or off actions regardless.

17:46:42 HTP: Command
17:46:42 SRC: WebCommand from 192.168.0.5
17:46:42 CMD: Group 0, Index 1, Command "POWER", Data "1"
17:46:42 RSL: stat/tasmota_D572A8/RESULT = {"POWER":"ON"}
17:46:42 RSL: stat/tasmota_D572A8/POWER = ON
17:46:42 RUL: POWER1#STATE performs "WebSend [192.168.0.5:39500] /?json={"StatusSTS":{"POWER1":"1"}}"
17:46:42 SRC: Rule
17:46:42 CMD: Group 0, Index 1, Command "WEBSEND", Data "[192.168.0.5:39500] /?json={"StatusSTS":{"POWER1":"1"}}"
**17:46:47 RSL: stat/tasmota_D572A8/RESULT = {"WebSend":"Connect failed"}**
17:46:47 CFG: Saved to flash at F7, Count 109, Bytes 4096
17:46:47 HTP: Command
17:46:47 SRC: WebCommand from 192.168.0.5
17:46:47 CMD: Group 0, Index 1, Command "POWER", Data "0"
17:46:47 RSL: stat/tasmota_D572A8/RESULT = {"POWER":"OFF"}
17:46:47 RSL: stat/tasmota_D572A8/POWER = OFF
17:46:47 RUL: POWER1#STATE performs "WebSend [192.168.0.5:39500] /?json={"StatusSTS":{"POWER1":"0"}}"
17:46:47 SRC: Rule
17:46:47 CMD: Group 0, Index 1, Command "WEBSEND", Data "[192.168.0.5:39500] /?json={"StatusSTS":{"POWER1":"0"}}"
**17:46:47 RSL: stat/tasmota_D572A8/RESULT = {"WebSend":"Done"}**

**Comment by hongtat** [comment]
The Tasmota device is unable to connect to ST hub (192.168.0.5:39500)
Check the connection between the Tasmota device and ST hub. It's a wifi/networking/firewall issue.

**Comment by dcuche** [comment]
> The Tasmota device is unable to connect to ST hub (192.168.0.5:39500)
> Check the connection between the Tasmota device and ST hub. It's a wifi/networking/firewall issue.

Hey thank you so much for your reply!
You are right about the 192.168.0.5 being the IP of the ST Hub... but... the issues is.. that I am controlling it with the ST hub... so.. for example, I press in the ST app to turn on the light, ST hub tells the device to turn on... it works.. and then after the fact the status update call from the tasmota to ST is the one that fails... but... it does not fails always... now I would say... it fails only 60% of the times maybe XD.... but ONLY when submiting the response... it NEVER fails when getting the initial order to turn on and off... so... that's why I think is weird.

One simple "patch" fix would be to reduce the timeout setting for the webcall to 1 seg (now its 5 seconds)... so it does not freeze the device for so long :S... (because when it works it just works in milliseconds)...

So... unstable networking?... but also weird because... without the tasmota connect up... it just works perfectly fine 100 times in a row controlling it over wifi..... 

anyway... thanks again!

**Comment by Boaz101** [comment]
I am having a similar issue. I have a SmartThings v2 hub and a Gosund sw1 wall switch running Tasmota 9.2. Sometimes the status update back to the hub succeeds and sometimes it doesn't. Seems almost random but it fails with the error WebSend: Connect failed more often than not. When it happens it delays the switch from updating it's led for 5s. Any ideas on what to look at? Perhaps I could capture the status update request somehow to make sure it's valid?

**Comment by dcuche** [comment]
> I am having a similar issue. I have a SmartThings v2 hub and a Gosund sw1 wall switch running Tasmota 9.2. Sometimes the status update back to the hub succeeds and sometimes it doesn't. Seems almost random but it fails with the error WebSend: Connect failed more often than not. When it happens it delays the switch from updating it's led for 5s. Any ideas on what to look at? Perhaps I could capture the status update request somehow to make sure it's valid?

Hi there, well in my case... I don't know if the problem solved itself (it seemed to be working properly more often every time ... or maybe its just me) but in between, I flashed and installed another Sonoff TX 2CH wall switch, and a couple of sonoff basics. By doing that I also ended up installing all the device handlers of the tasmota-connect app (as instructed... because I had not done so before, and just installed the one for the generic switch), and now everything works perfect! 👍  also updated the tasmota-connect app in the process (manually... I don't have it automatically connected to github).

and BTW.. I made no changes to (not even touched) any network/wifi nor firewall configurations.

so.. sorry I have no specifics on how it got solved... but just in case, I'm still using tasmota 9.1.0.2... and the tasmota connect says "1.1.1"

cheers!

**Comment by Boaz101** [comment]
Glad to hear that it's working for you. I already have all of the device handlers installed. I also have a static ip address set for my switch. I am on Tasmota Connect v1.1.2. Tasmota information says -63 dBm signal strength.

**Comment by kmlsmv1** [comment]
I have same problem, did someone find a solution?

**Comment by dcuche** [comment]
> I have same problem, did someone find a solution?

Real my commnet from Jan 4.... After some days with me just trying to forget about the issue... it seems to have solved itslef... for the Sonoff with 1 channel.
I also installed a Sonoff TX with 2 channels back then, and that one.... sometimes it does not work... it either send the signal to smarthings veeeery late (like almost a minute late).. or just timeouts completely and nothing happens.
The weird thing is that, my problem with the Sonoff TX 2CH is very different, because since I installed it, there has been like 2 periods of a couple weeks where it will ALWAYS be late or timeout.... during those 2 weeks... and after that time it solved itself.
Luckily, the first time I had that issue, I had purchased a generic zigbee button from aliexpress... and also a button from smartthings... so during those 2 weeks.. I just placed the button next to the light switch xD... and used that insted... testing the light switch every once in a while.. and well.. it solved itself again... but after a couple of months, the problem started again... lasted a couple of week... and now works again XD..
anyway... my point is... there DIY firmwares may not be that reliable, and it is better to have a backup way to handle the switches :)

Cheers!
